src/ar.erl

```diff
@@ -398,13 +409,20 @@ start(
 		]
 	),
 	%% Start the first node in the gossip network (with HTTP interface).
-	ar_http_iface_server:start(
-		Port,
-		Node,
-		SearchNode,
-		undefined,
-		Bridge
-	),
+	ar_http_iface_server:start(Port, [
+		{http_entrypoint_node, Node},
+		{http_search_node, SearchNode},
+		{http_bridge_node, Bridge}
+	]),
+	case GatewayOpts of
+		{on, GatewayPortString, GatewayDomainString} ->
+			GatewayPort = list_to_integer(GatewayPortString),
```

Comment:
Wouldn't it make sense if the `parse` function also did `list_to_integer` on `GatewayPort` and `list_to_binary ` on `GatewayDomain` and `GatewayCustomDomains`?

Reply:
Yep 👍